guacamole-ext/src/main/java/org/apache/guacamole/net/auth/UserContext.java

```diff
@@ -40,6 +40,23 @@ public interface UserContext {
      */
     User self();
 
+    /**
+     * Returns true if the session for the User associated with this user
+     * context is valid, or false otherwise. If the session is not valid, no
+     * user context actions may be peformed.
```

Comment:
I don't think we should commit to "no user context actions may be performed", as that would require more strict enforcement where we test `isValid()` before every action. I suggest instead only specifying that the user's session will be terminated by the web application.

It may also be useful to note here that the web application currently tests session validity every minute, though implementations should not rely on any particular schedule except that invalid sessions can be expected to be terminated in a reasonably timely manner.

Reply:
That makes sense - how does this look?

Reply:
Much better - LGTM!